lib/widgets/addTask.dart

```diff
@@ -3,232 +3,404 @@
+    setState(() {});
+  }
+
+  Widget buildName() {
+    return TextFormField(
+        autovalidateMode: AutovalidateMode.onUserInteraction,
+        controller: nameController,
+        decoration: InputDecoration(
+          contentPadding: const EdgeInsets.fromLTRB(16, 8, 0, 8),
+          border: const OutlineInputBorder(),
           hintText: 'Enter Task',
+          suffixIcon: nameController.text.isNotEmpty
+              ? IconButton(
+                  tooltip: "Clear",
+                  icon: const Icon(
+                    Icons.clear,
+                    size: 20,
+                  ),
+                  onPressed: () {
+                    setState(() {
+                      nameController.clear();
+                    });
+                  },
+                )
+              : null,
         ),
-        validator: (name) => name != null && name.isEmpty
-            ? 'You cannot leave this field empty!'
-            : null,
-      );
-  Widget buildPriority() => Column(children: [
-        Row(
-          mainAxisAlignment: MainAxisAlignment.start,
-          crossAxisAlignment: CrossAxisAlignment.center,
-          children: [
-            const Text(
-              'Priority :  ',
-              style: TextStyle(fontWeight: FontWeight.bold),
-              textAlign: TextAlign.left,
-            ),
-            DropdownButton<String>(
-              dropdownColor: AppSettings.isDarkMode
-                  ? const Color.fromARGB(255, 220, 216, 216)
-                  : Colors.white,
-              value: priority,
-              elevation: 16,
-              style: const TextStyle(color: Colors.black),
-              underline: Container(
-                height: 1.5,
-                color: Colors.black,
+        validator: (name) {
+          if (name == null || name.trim().isEmpty) {
+            return 'You cannot leave this field empty!';
+          }
+          return null;
+        });
+  }
+
+  Widget buildDueError(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.fromLTRB(16, 8, 0, 0),
+      child: dueError != null
+          ? Text(
+              dueError!,
+              style: Theme.of(context)
+                  .textTheme
+                  .bodySmall!
+                  .copyWith(color: Theme.of(context).colorScheme.error),
+            )
+          : null,
+    );
+  }
+
+  Widget buildDueDateTime() {
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Container(
+          constraints: const BoxConstraints(maxWidth: 130),
+          child: TextFormField(
+            autovalidateMode: AutovalidateMode.onUserInteraction,
+            controller: dateController,
+            onTap: () {
+              showDateDialog();
+            },
+            onChanged: (value) {
+              if (date == null) {
+                setState(() {
+                  dueError = "Due date & time can't be empty!";
```

Comment:
Instead of showing "Due date and time can't be empty", it will be better to make individual checks for the date and time and then display the relevant messages instead of a generalized message.

Reply:
Is this working completely with our model

Reply:
Yes it works. Creates a new Task on submission and passes due in the right DateTime format.